### PR TITLE
feat: used toHaveBeenCalledWith

### PR DESCRIPTION
**Pull request by lukeymyuan**
Replaced `mock.calls` with `toHaveBeenCalledWith` which is cleaner and more readable 

Co-authored-by: Conrad Lang <ConradLang@users.noreply.github.com>

### CODE REVIEW

**Review by samchungy — APPROVED** [review]

**Comment by 72636c** [comment]
Note that these are not equivalent. The previous code is exhaustive in asserting the total number of calls the logger receives, while the proposed code would allow additional logs to be emitted.

**Comment by samchungy** [comment]
> Note that these are not equivalent. The previous code is exhaustive in asserting the total number of calls the logger receives, while the proposed code would allow additional logs to be emitted.

Ah good point. I guess the equivalent would be toBeCalledTimes(2) but at that point not sure which is more cleaner. 3 v 1 expects

**Comment by lukeymyuan** [comment]
yeah.. i also thought about it...
let's just apply the changes where theres only 1 call

File: template/lambda-sqs-worker/src/framework/handler.test.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ describe('createHandler', () => {
 
     await expect(handler(input, ctx)).rejects.toThrow('Function failed');
 
-    expect(logger.error.mock.calls).toEqual([[{ err }, 'Function failed']]);
+    expect(logger.error).toHaveBeenCalledWith({ err }, 'Function failed');
```
Comment (trevorgk) [inline template/lambda-sqs-worker/src/framework/handler.test.ts:43]: Just a thought...
```suggestion
    expect(logger.error).toHaveBeenCalledTimes(1);
    expect(logger.error).toHaveBeenCalledWith({ err }, 'Function failed');
```

**Review by 72636c — APPROVED** [review]
Thanks, and sorry for the delay on this one! I think the proposed changes make sense; we care more that _an_ error log was written in these scenarios, not that it was the _only_ thing written.

**Pull request merged**